cclient/http/http.c

```diff
@@ -165,25 +229,45 @@ retcode_t iota_service_query(const void* const service_opaque,
       (const iota_client_service_t* const)service_opaque;
   const http_info_t* http_settings = &service->http;
 
-  if ((result = connect_to_iota_service(
-           http_settings->host, http_settings->port, &sockfd)) != RC_OK) {
-    goto cleanup;
-  }
+  if (http_settings->ca_pem == NULL) {
+    // HTTP
+    if ((result = connect_to_iota_service(
+             http_settings->host, http_settings->port, &sockfd)) != RC_OK) {
+      goto cleanup;
+    }
 
-  if ((result = send_headers_to_iota_service(sockfd, http_settings,
-                                             obj->length)) != RC_OK) {
-    goto cleanup;
-  }
-  if ((result = send_data_to_iota_service(sockfd, obj->data, obj->length)) !=
-      RC_OK) {
-    goto cleanup;
-  }
-  if ((result = read_data_from_iota_service(sockfd, response)) != RC_OK) {
-    goto cleanup;
-  }
-cleanup:
-  if (sockfd != -1) {
-    close_socket(sockfd);
+    if ((result = send_headers_to_iota_service(sockfd, http_settings,
```

Comment:
Can we uniform names between http and https ?
- `connect_to_iota_service ` and `tls_socket_connect `
- `close_socket `  and  `tls_socket_close `
- `send_headers_to_iota_service ` and `https_request_header `
- `send_data_to_iota_service ` and `https_request_data `
- `read_data_from_iota_service `and `https_response_read `